src/platform/plugins/shared/discover/public/application/main/state_management/redux/actions/tabs.ts

```diff
@@ -49,13 +49,22 @@ export const setTabs: InternalStateThunkActionCreator<
     { runtimeStateManager, tabsStorageManager, services: { profilesManager, ebtManager } }
   ) {
     const previousState = getState();
+    const discoverSessionChanged =
+      params.updatedDiscoverSession &&
+      previousState.persistedDiscoverSession &&
+      params.updatedDiscoverSession.id !== previousState.persistedDiscoverSession?.id;
```

Comment:
We now support passing an updated Discover session directly through `setTabs`. If it's different from the one we currently have (such as when copying on save), we automatically remove all existing tabs and re-add the updated ones to reset the runtime states. This keeps changes to the runtime tab state map consolidated, and ensures it's always in sync with the Redux store. A different session ID with the same tab IDs shouldn't really happen now anyway since we're regenerating the IDs when copying on save, but this approach would work regardless.